applications/nrf_desktop/src/modules/usb_state_pm.c

```diff
@@ -58,7 +72,23 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_ALIVE);
 			break;
 		case USB_STATE_DISCONNECTED:
-			power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_MAX);
+			if (CONFIG_DESKTOP_USB_PM_RESTRICT_REMOVE_DELAY_MS) {
+				/* Remove the restriction after the delay to allow the user to
+				 * take actions like e.g. restart advertising without going
+				 * through system off.
+				 */
+				k_work_schedule(
+					&pm_restrict_remove_work,
+					K_MSEC(CONFIG_DESKTOP_USB_PM_RESTRICT_REMOVE_DELAY_MS));
+			} else {
+				power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_MAX);
+			}
+			/* When disconnecting USB cable, the USB_STATE_SUSPENDED event may
+			 * or may not appear before the USB_STATE_DISCONNECTED event. Force power
+			 * down for application to behave consistently even if the
+			 * USB_STATE_SUSPENDED event doesn't appear.
+			 */
+			force_power_down();
```

Comment:
What do you think about separating 2 small functions (for force power down handling and applying power manager restrictions - both based on USB state)? Similarly to `update_pm_policy_latency_req`.

That might improve readability here I think (as @pdunaj mentioned, the function is already quite complex)